### PR TITLE
chore: impl report telemetry to greptime

### LINKED ISSUE #1587: GreptimeDB Telemetry
### What type of enhancement is this?

Other

### What does the enhancement do?

We need to add the telemetry to greptimedb, which collects the minimum info from the user's download or build binaries. Just like tidb telemetry https://github.com/pingcap/tidb/tree/master/telemetry

Some info may be useful:

* The GreptimeDB version
* the number of cluster nodes

The telemetry should be disabled by default, and we have to provide a document to describe it like https://docs.pingcap.com/tidb/dev/telemetry

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by paomian**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

* impl telemetry reporter to statistic

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

close #1587

### CODE REVIEW

File: src/common/runtime/Cargo.toml
Comment (MichaelScofield) [inline src/common/runtime/Cargo.toml:1]: I think we'd better place version report codes in crate `common-meta` instead. `common-runtime` has nothing to do with reporting.

Comment (MichaelScofield) [reply src/common/runtime/Cargo.toml:1]: Or a new crate `telemetry` maybe.

**Comment by killme2008** [comment]
@paomian Please rebase the branch and trigger the CI again, there are some errors.

**Comment by paomian** [comment]
I will close this pr, and create new one later.

**Pull request closed without merging**